webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -213,20 +214,23 @@ class PolicyWriterServiceImpl(
   override def writeTemplate(
       rootNodeId      : NodeId
     , nodesToWrite    : Set[NodeId]
-    , allNodeConfigs  : Map[NodeId, NodeConfiguration]
+    , nodeConfigsToWrite  : Map[NodeId, NodeConfiguration]
+    , allNodeInfos    : Map[NodeId, NodeInfo]
```

Comment:
ok, this part may be useful, as there's some memory constraint right before Promises prepared in xxx ms